### DISCUSSION TITLE
How browsersync manages to scroll web and mobile view in sync while scrolling

### DISCUSSION THREAD

**Discussion by basit-skill-lync** [BrowserSync/browser-sync#2012] [Q&A]
Want to know the logic how browsersync is managing sync between web and mobile view while scrolling a webpage.

**Comment by shakyShane** [comment]
Just think about it like a multiplayer game.

All devices are 'connected' via web sockets. When you scroll on one device, that one temporarily becomes the 'controller' in a sense. 

So, whilst scrolling on that device, it's constantly emitting it's Y position (well, a percentage of the page scrolled) and other devices simply apply this :)

**Reply by basit-skill-lync** [reply]
Thanks for the response @shakyShane ,
I am currently working on a requirement where the admin can scroll the content of all the connected clients. 
I am able to achieve that using socket and emitting the 'y' coordinate but the problem is coming where the admin is on web and the client is on mobile , there sync is not happening because of web-mobile ratio is not same.
Can you share some snippet or elaborate how percentage based scrolling can be achieved on different devices.

**Reply by basit-skill-lync** [reply]
Hi @shakyShane  , want to know, Are you emitting every event that is happening on the DOM like scroll, click etc, using socket.io.
 Or are you using any other mechanism other than socket like screen share ?
